File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/PupilRepository.java
package com.example.schoolsite.workWithDatabase.repo;

import com.example.schoolsite.entity.Pupil;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PupilRepository extends JpaRepository<Pupil, Long> {
    Pupil findByNameAndLastnameAndPatronymic(String name, String lastname, String patronymic);
    Pupil findByUserId(Long userId);
    List<Pupil> findAllByClassroomId(Long classroomId);
}

File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/PupilRepository.kt
package com.example.schoolsite.workWithDatabase.repo

import com.example.schoolsite.entity.Pupil
import org.springframework.data.jpa.repository.JpaRepository
import java.util.List
import org.springframework.stereotype.Repository

@Repository
interface PupilRepository : JpaRepository<Pupil?, Long?> {
    fun findByNameAndLastnameAndPatronymic(name: String?, lastname: String?, patronymic: String?): Pupil?
    fun findByUserId(userId: Long?): Pupil?
    fun findAllByClassroomId(classroomId: Long?): List<Pupil?>?
}